core/common/src/main/java/alluxio/client/quota/CacheScope.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.quota;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * This class defines "Scope" of a cache quota. For Presto caching, scope is
+ * hierarchical with different levels including:
+ * <ul>
+ *   <li>GLOBAL (".") for global scope;</li>
+ *   <li>SCHEMA (e.g., "schema1") for an entire schema;</li>
+ *   <li>TABLE (e.g., "schema2.table1") for a given table;</li>
+ *   <li>PARTITION (e.g., "schema3.table2.partition1") for a given partition</li>
+ * </ul>
+ */
+@ThreadSafe
+public class CacheScope {
+  private static final String GLOBAL_ID = ".";
+  public static final String SEPARATOR = ".";
+  public static final CacheScope GLOBAL = new CacheScope(GLOBAL_ID, 1, Level.GLOBAL);
+
+  /**
+   * Level of a scope.
+   */
+  public enum Level {
```

Comment:
Do we need to use regex here or can we use split by delimiter and index?

Reply:
I'd like to keep it cleaner & more flexible initially.
In case it consumes too much cpu in profiling, we can change it to string split.